scripts/h5peditor-image.js

```diff
@@ -173,6 +244,202 @@ ns.widgets.image.prototype.appendTo = function ($wrapper) {
+    const filesBackup = [new File([files[0]], files[0].name, { type: files[0].type })];
+    this.replaceCb = () => {
+      this.removeImage();
+      this.uploadFile(filesBackup);
+    };
+    this.previousState = this.STATE.HAS_IMAGE;
+    this.previousParams = this.params;
+    this.confirmReplaceDialog.show(this.$imageDropContainer.offset().top);
+  } else {
+    this.previousState.NO_IMAGE;
+    this.previousParams = null;
+    this.uploadFile(files);
+  }
+};
+
+/**
+ * Add drag and drop listeners to the appropriate DOM elements.
+ */
+ns.widgets.image.prototype.addDragAndDropListeners = function () {
+  const boundedHandleDnD = this.handleDragAndDrop.bind(this);
+
+  this.dropBlockEl = this.boxEl.querySelector('.h5p-dnd__box__block');
+  this.boxEl.addEventListener('dragenter', boundedHandleDnD);
+  this.dropBlockEl.addEventListener('dragover', boundedHandleDnD);
+  this.dropBlockEl.addEventListener('dragend', boundedHandleDnD);
+  this.dropBlockEl.addEventListener('dragleave', boundedHandleDnD);
+  this.dropBlockEl.addEventListener('drop', boundedHandleDnD);
+};
+
+/**
+ * Upload the first file in the FileList.
+ * @todo: handle multiple files?
+ * 
+ * @param {FileList} files 
+ */
+ns.widgets.image.prototype.uploadFile = function (files) {
+  if (files.length > 0) {
+    this.upload(files[0], files[0].name);
+  }
+};
+
+/**
+ * Handle drag and drop events. Will apply the correct css classes (styling)
+ * depending on event type and will handle uploading if the user drops the file.
+ *
+ * @param {Event} e 
+ */
+ns.widgets.image.prototype.handleDragAndDrop = function (e) {
+  e.preventDefault();
+  e.stopPropagation();
+
+  const container = e.target.closest('.h5p-dnd__container');
+  const boxBlock = container.querySelector('.h5p-dnd__box__block');
+  
+  if (e.type === 'dragenter') {
+    this.boxEl.classList.add('h5p-dnd__box--is-dragging')
+  }
+  else if (e.type === 'dragend' || (e.type === 'dragleave' && boxBlock === e.target)) {
+	  this.boxEl.classList.remove('h5p-dnd__box--is-dragging');
+  }
+  else if (e.type === 'drop') {
+    this.boxEl.classList.remove('h5p-dnd__box--is-dragging');
+    this.uploadOrReplaceImage(e.dataTransfer.files);
+  }
+};
+
+/**
+ * Generates the HTML for the NO_IMAGE state.
+ * 
+ * @returns string The HTML for the NO_IMAGE state
+ */
+ns.widgets.image.prototype.getBaseMarkup = function () {
+  return `
+    <div class="h5p-dnd__box__block"></div>
+    <div class="h5p-dnd__column h5p-dnd__column--is-highlighted h5p-dnd__column--is-fixed h5p-dnd__column--hide-when-focus">
+      ${this.getImageUploadSvg()}
+      <button class="h5p-dnd__btn h5p-dnd__btn__upload" type="button">${ns.t('core', 'uploadImage')}</button>
+    </div>
+
+    <div class="h5p-dnd__column h5p-dnd__column--hide-when-focus">
+      <div>
+        ${ns.t('core', 'dragAndDropAndPasteImage')} <span class="h5p-dnd__pill">ctrl</span> + <span class="h5p-dnd__pill">v</span>
```

Comment:
@wwalmnes Just wonder if this should be aware of the operating system: On MacOS, pasting is done using `Command` + `V` instead. Ask Jelena maybe :-)

Reply:
Hey @otacke, the ⌘ ascii will be added indeed :)